[s] Makes used grenades slightly more obvious. Also prevents DOSing admins.

TemporalOroboros opened the pull request:
Adds a variable to tell whether a grenade can't detonate properly.
Examining can be used to tell if the grenade has already been used.
A flag on it is set when the grenade detonates so that used grenades can't be reused until it's reset.
It also prevents spamming admins with grenade primed messages since there's no point if the grenade won't work.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds a variable to track whether a grenade has already been used.
If it has it prevents the grenade from detonating again until the grenade has been reset. (Resetting only implemented for chemical grenades.)
It also prevents priming the grenade/pulsing the grenades wires from messaging admins (but still lets them log) because there's no point in telling the admins if the grenade isn't going to do anything.
The status of the grenade shows up in the examine text.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
1. Makes it slightly more obvious that the cleaner grenades in the halls have already been used.
2. Prevents DOSing admins with grenade logging spam. (Fixes #64640)

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
qol: Used grenades now look used when examined.
fix: It should no longer be possible to DC admins by spamming grenades.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TemporalOroboros committed:
Makes used grenades slightly more obvious.

Adds a variable to tell whether a grenade can't detonate properly.
Examining can be used to tell if the grenade has already been used.
A flag on it is set when the grenade detonates so that used grenades can't be reused until it's reset.
It also prevents spamming admins with grenade primed messages since there's no point if the grenade won't work.

TheVekter commented:
Based, this should prevent this from happening.

![image](https://user-images.githubusercontent.com/7019927/152405257-89151d1c-a0d0-4b53-80b6-90b57a3e9407.png)

tralezab reviewed (changes requested):
the rest looks good

TemporalOroboros committed:
Changes dud var to dud flags.

Changes the name of the dud var to dud_flags to clarify that it is a set of bitfields.
Changes the default value of the dud var to NONE also to clarify that it is a set of bitfields.
Changes the documentation for the dud var to clarify that it is a set of bitfields.

TemporalOroboros committed:
Fixes eternally armed dud grenades.

Makes the grenades reset their appearance if they reach the end of their fuse while a dud.
Prevents chemical grenades from staying in the 'i am armed' visual state forever.

TemporalOroboros commented on code/game/objects/items/grenades/_grenade.dm:
```diff
@@ -17,6 +23,8 @@
 	resistance_flags = FLAMMABLE
 	max_integrity = 40
+	/// Whether this grenade can detonate.
+	var/dud = FALSE
```
Done.

Elli-Skala commented:
This takes into account the advanced release grenade right?
